qiskit/algorithms/state_fidelities/compute_uncompute.py

```diff
@@ -0,0 +1,118 @@
+
+        where :math:`x` and :math:`y` are optional parametrizations of the
+        states :math:`\psi` and :math:`\phi` prepared by the circuits
+        ``circuit_1`` and ``circuit_2``, respectively.
+        Args:
+            sampler: Sampler primitive instance.
+            run_options: Backend runtime options used for circuit execution.
+        """
+        self._sampler = sampler
+        self._default_run_options = run_options
+        super().__init__()
+
+    def create_fidelity_circuit(self, circuit_1, circuit_2) -> QuantumCircuit:
+        """
+        Creates fidelity circuit following the compute-uncompute method.
+        Args:
+            circuit_1: (Parametrized) quantum circuit
+            circuit_2: (Parametrized) quantum circuit
+
+        Returns:
+            The fidelity quantum circuit corresponding to circuit_1 and circuit_2.
+        """
+        circuit = circuit_1.compose(circuit_2.inverse())
+        circuit.measure_all()
+        return circuit
+
+    def _run(
+        self,
+        circuits_1: Sequence[QuantumCircuit],
+        circuits_2: Sequence[QuantumCircuit],
+        values_1: Sequence[Sequence[float]] | None = None,
+        values_2: Sequence[Sequence[float]] | None = None,
+        **run_options,
+    ) -> StateFidelityResult:
+        r"""
+        Compute the state overlap (fidelity) calculation between two
+        (parametrized) circuits (first and second) for a specific set of parameter
+        values (first and second) following the compute-uncompute method
+
+        The fidelity corresponds to:
+
+            :math:`|\langle\psi(x)|\phi(y)\rangle|^2`
+
+        Args:
+            circuits_1: (Parametrized) quantum circuits preparing :math:`|\psi\rangle`.
+            circuits_2: (Parametrized) quantum circuits preparing :math:`|\phi\rangle`.
+            values_1: Numerical parameters to be bound to the first circuits.
+            values_2: Numerical parameters to be bound to the second circuits.
+            run_options: Backend runtime options used for circuit execution.
+
+        Returns:
+            The result of the fidelity calculation.
+
+        Raises:
+            ValueError: At least one pair of circuits must be defined.
+        """
+
+        circuits = self._construct_circuits(circuits_1, circuits_2)
+        values = self._construct_value_list(circuits_1, circuits_2, values_1, values_2)
+
+        if len(circuits) == 0:
+            raise ValueError(
+                "At least one pair of circuits must be defined to calculate the state overlap."
+            )
+
+        # The priority of run options is as follows:
+        # run_options in `evaluate` method > fidelity's default run_options >
+        # primitive's default run_options.
+        run_opts = copy(self._default_run_options)
+        run_opts.update(**run_options)
+
+        job = self._sampler.run(circuits=circuits, parameter_values=values, **run_opts)
+
+        result = job.result()
```

Comment:
Shouldn't we be checking the status of the job and managing accordingly if its not `DONE`